include/llbuild/Basic/RedirectedProcess.h

```diff
@@ -0,0 +1,42 @@
+//===- RedirectedProcess.h --------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_REDIRECTEDPROCESS_H
+#define LLBUILD_BASIC_REDIRECTEDPROCESS_H
+
+#include <signal.h>
+
+namespace llbuild {
+namespace basic {
+class RedirectedProcess {
+ public:
+  using ProcessId = pid_t;
+  ProcessId processId;
+
+  RedirectedProcess(ProcessId processId) { this->processId = processId; }
+
+  void kill(int signal);
+
+  struct Hash {
+    size_t operator()(const llbuild::basic::RedirectedProcess& x) const {
+      return x.processId;
+    }
+  };
+
+  bool operator==(const RedirectedProcess& rhs) const {
```

Comment:
Is there a reason I would want it to be equatable versus working with pointers to it?

This (and the hash function) are necessary for it to be contained in an `unordered_set`. Maybe we could use a simple `llvm::ArrayRef` to hold `RedirectedProcess`es though @ddunbar?